📖 Add pivoting and provider installation instructions (#377)
metal3-io/cluster-api-provider-metal3: README.md

| v1alpha5      | v1alpha4            | v0.5.X        |
| v1beta1       | v1beta1             | v1.0.X        |

## Deploying the metal3 provider

The recommended method is using
[Clusterctl](https://master.cluster-api.sigs.k8s.io/clusterctl/overview.html).

Starting from `v0.5.0` release of Cluster-api-provider-metal3, Baremetal Operator is decoupled
from Cluster-api-provider-metal3 deployments when deployed via `clusterctl`. For that reason,
Baremetal Operator will not be installed when initializing the Metal3 provider with clusterctl,
and its CRDs and controller need to be manually installed. Example flow of installing Metal3
provider:

1. Install Cluster API core, bootstrap and control-plane providers. This will also install
  cert-manager if it is not already installed.

    ```shell
    clusterctl init --core cluster-api:v0.4.4 --bootstrap kubeadm:v0.4.4 \
        --control-plane kubeadm:v0.4.4 -v5
    ```

> 1. Install Metal3 provider. This will install Cluster-api-provider Metal3 CRDs and controllers

Reviewer: ```suggestion
1. Install Metal3 provider. This will install Cluster-api-provider Metal3 CRDs and controllers.
```

    ```shell
    clusterctl init --infrastructure metal3
    ```

> 1. Install Baremetal Operator and its CRD. You need to install cert-manager for Baremetal Operator,

Reviewer: ```suggestion
1. Deploy Baremetal Operator manifests and CRDs. You need to install cert-manager for Baremetal Operator,
```

    cd baremetal-operator
    kustomize build config/default | kubectl apply -f -
    ```

1. Install Ironic. There are a couple of ways to do it.
    - Run within a Kubernetes cluster as a pod, refer to [deploy.sh](https://github.com/metal3-io/baremetal-operator/blob/master/tools/deploy.sh)
      script.
    - Outside of a Kubernetes cluster as a container. Please refer to [run_local_ironic.sh](https://github.com/metal3-io/baremetal-operator/blob/master/tools/run_local_ironic.sh).

Please refer to the [getting-started](docs/getting-started.md) for more info.

## Pivoting ⚠️

Starting from `v0.5.0` release of Cluster-api-provider-metal3, Baremetal Operator is decoupled
from Cluster-api-provider-metal3 deployments when deployed via `clusterctl`. For that reason,
when performing `clusterctl move`, custom objects outside of the Cluster API chain or not part
of CAPM3 will not be pivoted to a target cluster. Example to those objects is BareMetalHost, its
secret and configMap which are reconciled by Baremetal Operator. To ensure that those objects are
also pivoted as as part of `clusterctl move`, `clusterctl.cluster.x-k8s.io` label need to be set

> on the BareMetalHost CRD before pivoting.

Reviewer: ```suggestion
on the BareMetalHost CRD before pivoting. If other CRDs also need to be pivoted to target cluster we need to add the same label in the CRDs. 
```

| v1alpha5      | v1alpha4            | v0.5.X        |
| v1beta1       | v1beta1             | v1.0.X        |

## Deploying the metal3 provider

The recommended method is using

> [Clusterctl](https://master.cluster-api.sigs.k8s.io/clusterctl/overview.html).

Reviewer 2: please revert this back, old URL seems right already
